lib/active_admin/resource_controller/data_access.rb

```diff
@@ -127,7 +128,10 @@ def build_resource
       #
       # @return [ActiveRecord::Base] An un-saved active record base object
       def build_new_resource
-        scoped_collection.send method_for_build, *resource_params
+        scoped_collection.send(
+          method_for_build,
+          *resource_params.map { |params| params.slice(active_admin_config.resource_class.inheritance_column) }
```

Comment:
@faucct  I have many Form objects in AA so I would like to see it like 
```
scoped_collection.send( method_for_build, resource_params_for_build)
```

and 
```
def  resource_params_for_build
   if active_admin_config.resource_class.respond_to?(:inheritance_column)
      params.slice(active_admin_config.resource_class.inheritance_column)
  else 
     {}
   end
end
```

Reply:
The solution you are suggesting would be valid in case you register a non-activerecord model. Is that what you do? Can you show me an example, so I could come up with test?

Reply:
@faucct  sort of this http://kenglish.co/using-form-pattern-with-an-activeadmin-controller/

Reply:
Are you sure that a workaround to support a specific case of registering a non-activerecord resource belongs to activeadmin? You could fix this by defining an empty `inheritance_column` method in your form.

Reply:
@faucct  I already approved PR it is only one more suggestion to improve

Reply:
So, merge?

Reply:
ping

Reply:
@faucct  2 review needed for merge 
cc @varyonic